### PR TITLE
Checkpoint limits

### PR DESCRIPTION
**Pull request by mootz12**
Governor
* update to use ledgers instead of seconds
* set vote ledgers when proposal is created

Votes
* update checkpoint system to only store relevant checkpoints based on future vote ledgers
* use ledgers instead of seconds

### CODE REVIEW

**Review by Ryang-21 — APPROVED** [review]
Neat fix for the checkpoint tracking! Had a type in a function definition but everything else lgtm.

**Review by Ryang-21 — COMMENTED** [review]
Neat fix for the checkpoint tracking! Had a typo in a function definition but everything else lgtm.

**Review by Ryang-21 — COMMENTED** [review]
Neat fix for the checkpoint tracking! Had a type in a function definition but everything else lgtm.

File: Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ codegen-units = 1
 lto = true
 
 [workspace.dependencies.soroban-sdk]
-version = "20.1.0"
+version = "20.0.0"
```
[the remark this answers is not in the document]
Comment (mootz12) [reply Cargo.toml:26]: ah i was trying to debug something. I updated it to current

**Review by Ryang-21 — APPROVED** [review]
lgtm

**Pull request merged**